build-logic/convention/build.gradle.kts

```diff
@@ -33,4 +33,7 @@ dependencies {
     implementation(libs.kotlin.serializationPlugin)
     implementation(libs.google.devtools.kspPlugin)
     implementation(libs.metro.gradlePlugin)
+
+    // TODO: https://github.com/gradle/gradle/issues/15383
+    implementation(files(libs.javaClass.superclass.protectionDomain.codeSource.location))
```

Comment:
Given this is a workaround, and may break in newer gradle versions, is it safer to just jeep using the non type-safe strings for now?

Reply:
I believe this workaround is stable, but either way is good to me. 

I addressed this PR here as I see the dependencies like `lifecycle-runtime` are not marked as referenced in IDEA, as they are string refs.

Reply:
Yeah makes sense, makes it easier to identify unused dependencies.